src/main/kotlin/io/inventi/eventstore/publisher/EventPublisher.kt

```diff
@@ -0,0 +1,49 @@
+package io.inventi.eventstore.publisher
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.msemys.esjc.EventData
+import com.github.msemys.esjc.EventStore
+import com.github.msemys.esjc.ExpectedVersion
+import io.inventi.eventstore.util.LoggerDelegate
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import kotlin.concurrent.thread
+
+/**
+ * Serializes and appends given object to configured stream in EventStore.
+ */
+@Service
+class EventPublisher {
+
+    private val logger by LoggerDelegate()
+
+    @Value("\${eventstore.eventPublisher.streamName}")
+    private lateinit var streamName: String
+
+    @Autowired
+    private lateinit var objectMapper: ObjectMapper
+
+    @Autowired
+    private lateinit var eventStore: EventStore
+
+    fun <T> publish(eventType: String, eventData: T) : Unit {
+        thread() {
```

Comment:
Creating a new thread is quite expensive on the JVM. Instead of creating a thread per event message, we should use some sort of thread pool or a coroutine. Coroutines might be an overkill here, so we could use `val pool = Executors.newFixedThreadPool(...) ; pool.execute { }`